bin/is-website-vulnerable.js

```diff
@@ -6,91 +6,87 @@ const os = require('os')
+
+        try {
+          const response = await prompt({
+            type: 'input',
+            name: 'url',
+            message: 'Please provide a URL to scan:',
+            validate: input => input && input.length > 0 && isUrl(input)
+          })
+          url = response.url
+        } catch (_) {
+          throw new Error('URL input prompt failed')
+        }
       }
     }
-  }
 
-  url = Utils.parseUrl(url)
+    if (!isUrl(url)) {
+      throw new Error('Given argument is an invalid URL')
+    }
 
-  if (isWindows && !isJson) {
-    console.log('Please wait, scanning the website (can take up to a minute)...')
-  }
+    url = Utils.parseUrl(url)
 
-  debug(`detecting isWindows: ${isWindows}`)
-  debug(`detecting isJson: ${isJson}`)
-  debug(`showing progress bar: ${isWindows}`)
+    if (isWindows && !isJson) {
+      console.log('Please wait, scanning the website (can take up to a minute)...')
+    }
 
-  const opts = {
-    lighthouseOpts: {},
-    chromeOpts: {}
-  }
+    debug(`detecting isWindows: ${isWindows}`)
+    debug(`detecting isJson: ${isJson}`)
+    debug(`showing progress bar: ${isWindows}`)
 
-  if (Utils.hasDevice(argv)) {
-    opts.lighthouseOpts = Object.assign(opts.lighthouseOpts, {
-      emulatedFormFactor: Utils.parseDevice(argv)
-    })
-  }
+    const opts = {
+      lighthouseOpts: {},
+      chromeOpts: {}
+    }
 
-  const chromePath = argv.chromePath
-  if (chromePath) {
-    opts.chromeOpts = Object.assign(opts.chromeOpts, { chromePath })
-  }
+    if (Utils.hasDevice(argv)) {
+      opts.lighthouseOpts = Object.assign(opts.lighthouseOpts, {
+        emulatedFormFactor: Utils.parseDevice(argv)
+      })
+    }
 
-  const audit = new Audit()
+    const chromePath = argv.chromePath
+    if (chromePath) {
+      opts.chromeOpts = Object.assign(opts.chromeOpts, { chromePath })
+    }
 
-  let results
-  // eslint-disable-next-line no-useless-catch
-  try {
-    results = await audit.scanUrl(url, opts, showProgressBar)
-  } catch (error) {
-    throw error
-  }
-  if (results.lhr.runtimeError) {
-    throw new Error(results.lhr.runtimeError.message)
-  }
+    const audit = new Audit()
 
-  let renderer
-  if (argv.json) {
-    renderer = new RenderJson(results, argv.jsLib)
-  } else {
-    renderer = new RenderConsole(results, argv.jsLib)
-  }
-  renderer.print()
+    const results = await audit.scanUrl(url, opts, showProgressBar)
+    if (results.lhr.runtimeError) {
+      throw new Error(results.lhr.runtimeError.message)
+    }
 
-  process.exit(0)
-}
+    let renderer
+    if (argv.json) {
+      renderer = new RenderJson(results, argv.jsLib)
+    } else {
+      renderer = new RenderConsole(results, argv.jsLib)
+    }
+    renderer.print()
 
-;(async () => {
-  try {
-    await main()
+    process.exit(0)
   } catch (error) {
-    console.error(`\nError: ${error.message}\n`)
+    console.error(`\n${chalk.red.underline('Error')}: ${error.message}\n`)
```

Comment:
@itaisteinherz can we remove the colors here?
if this runs on CI it could prove hard to read

Reply:
Sure 👌🏻

Reply:
Done

Reply:
thanks!